nixos/modules/services/logging/logrotate.nix

```diff
@@ -36,15 +33,17 @@ let
         type = with types; nullOr str;
         default = null;
         description = ''
-          The user account to use for rotation.
+          The user account to use for rotation. This setting is deprecated, please use
```

Comment:
If that settings is to be deprecated I would use `mkRemovedOptionModule` with a message indicating to use `su`, because a simple change in the description goes easily unnoticed.
And simplify the module and test by removing the new code to support the backward compatibility.

Reply:
I wanted to leave a window with the setting deprecated but still working, it's possible by adding warnings by setting config.warnings conditionally (plenty of examples about) but I didn't get around to it yet.

I agree the description isn't enough in itself, but I think warnings can be added in a subsequent PR so as not to slow this down -- I don't really like PRs growing in scope all the time and prefer to keep goals somewhat constrainted... But if I get around to it fast enough I'll update this PR when I'm done.

Reply:
(that being said, if the consensus is to deprecate settings the hard way, it'll definitely make this module a bit simpler and I don't particularly mind for myself. We can easily remove everywhere I've marked deprecated and most of the exceptions I added to generateConfig)